contrib/endpoints/src/api_manager/service_control/proto_test.cc

```diff
@@ -161,6 +173,70 @@ TEST_F(ProtoTest, FillGoodCheckRequestTest) {
   ASSERT_EQ(expected_text, text);
 }
 
+TEST_F(ProtoTest, FillGoodAllocateQuotaRequestTest) {
+  std::vector<std::pair<std::string, int>> metric_cost_vector;
+  metric_cost_vector.push_back(
+      std::make_pair<std::string, int>("metric_first", 1));
+  metric_cost_vector.push_back(
+      std::make_pair<std::string, int>("metric_second", 2));
+
+  google::api_manager::service_control::QuotaRequestInfo info;
+
+  info.metric_cost_vector = &metric_cost_vector;
+
+  FillOperationInfo(&info);
+  FillAllocateQuotaRequestInfo(&info);
+
+  gasv1::AllocateQuotaRequest request;
+  ASSERT_TRUE(scp_.FillAllocateQuotaRequest(info, &request).ok());
+
+  std::string text = AllocateQuotaRequestToString(&request);
+  std::string expected_text = ReadTestBaseline("allocate_quota_request.golden");
+  ASSERT_EQ(expected_text, text);
+}
+
+TEST_F(ProtoTest, FillAllocateQuotaRequestNoMethodNameTest) {
+  std::vector<std::pair<std::string, int>> metric_cost_vector;
+  metric_cost_vector.push_back(
+      std::make_pair<std::string, int>("metric_first", 1));
+  metric_cost_vector.push_back(
+      std::make_pair<std::string, int>("metric_second", 2));
+
+  google::api_manager::service_control::QuotaRequestInfo info;
+
+  FillOperationInfo(&info);
+  info.metric_cost_vector = &metric_cost_vector;
+  info.client_ip = "1.2.3.4";
+  info.referer = "referer";
+  info.method_name = "";
+
+  gasv1::AllocateQuotaRequest request;
+  ASSERT_TRUE(scp_.FillAllocateQuotaRequest(info, &request).ok());
+
+  std::string text = AllocateQuotaRequestToString(&request);
+  std::string expected_text =
+      ReadTestBaseline("allocate_quota_request_no_method_name.golden");
+  ASSERT_EQ(expected_text, text);
+}
+
+TEST_F(ProtoTest, FillNoMetricRulesAllocateQuotaRequestTest) {
+  google::api_manager::service_control::QuotaRequestInfo info;
+
+  info.metric_cost_vector = nullptr;
```

Comment:
If a method have nullptr for metric_cost_vector,  quota call is NOT needed. we don't need such test case.  We don't want to call allocateQuota without any metrics.